version: "3.8"
services:
  application:
    build: .
    container_name: application
    restart: on-failure
    ports:
      - 8081:8081
    environment:
      SPRING_APPLICATION_JSON: '{
        "spring.datasource.url"  : "jdbc:mysql://mysqldb:3306/clinica_veterinaria_db?createDatabaseIfNotExist=true&allowPublicKeyRetrieval=true&useSSL=false",
        "spring.datasource.username" : "root",
        "spring.datasource.password" : "1234",
        "spring.datasource.driver-class-name": "com.mysql.cj.jdbc.Driver",
        "spring.jpa.database-platform" : "org.hibernate.dialect.MySQL5InnoDBDialect",
        "spring.jpa.generate-ddl": "true",
        "spring.jpa.hibernate.ddl-auto" : "update",
        "spring.servlet.multipart.max-file-size": "10MB",
        "spring.servlet.multipart.max-request-size": "10MB",
        "spring.servlet.multipart.enabled" : "true"
      }'
    networks:
      - clinica-veterinaria-net
    depends_on:
      mysqldb:
        condition: service_healthy
  mysqldb:
    image: mysql:8
    container_name: mysqldb
    restart: unless-stopped
    volumes:
      - db:/var/lib/mysql
    networks:
      - clinica-veterinaria-net
    environment:
      - MYSQL_ROOT_PASSWORD=1234
      - MYSQL_DATABASE=clinica_db
    healthcheck:
      test: ["CMD", "mysqladmin" ,"ping", "-h", "localhost"]
      timeout: 10s
      retries: 5
    ports:
      - 3306:3306
networks:
  clinica-veterinaria-net:
     driver: bridge
volumes:
  db:
